rfd/0194-selinux-policy.md

```diff
@@ -0,0 +1,150 @@
+When an SSH session is started, the SSH service will create 2 child processes to handle it: 
+
+- An intermediary process that prepares another child process that will actually run the SSH session. Depending on what is configured, PAM contexts
+may be created, users may be created, a shell will be created, the user and group will be changed if necessary, and so on.
+- The intermediary process will then create the child session process, which only needs standard permissions of the host user that the SSH session
+was created for.
+
+Additionally, the SSH agent may create a networking child process to handle X11 forwarding, port forwarding, and SSH Agent forwarding. This process
+is a child process of the intermediary process as well.
+
+4 main SELinux domains will be created to ensure each process in the chain is granted only the necessary permissions: `teleport_ssh_service`,
+`teleport_ssh_service_intermediary`, `teleport_ssh_service_session`, and `teleport_ssh_service_networking` respectively.
+
+Configurable file and port contexts will be created for the `teleport_ssh_service` domain, and configurable file contexts
+will be created for the `teleport_ssh_service_intermediary` domain to allow those processes to bind to the necessary ports and read and write
+to the necessary files, depending on the SSH agent configuration. At a minimum this will allow the `teleport_ssh_service` domain to bind to
+the configured SSH port and read and write to the data directory, and dial the configured Teleport Proxy and Auth servers. 
+
+There are a few optional features that can be enabled or disabled on SSH agents, the first version of the policy will ensure necessary permissions
+for all optional features are allowed to simplify development and deployment.
+
+These include:
+
+- Reading `~/.tsh/environment` on the server before creating a session
+    - The `teleport_ssh_service_intermediary` domain will be allowed to read `.tsh/environment` in all users' home directories
+- Creation of host users and groups
+    - The `teleport_ssh_service` domain will be allowed to call necessary binaries such as `useradd`, `usermod` and `groupadd`
+- Enhanced session recording with BPF
+   - The `teleport_ssh_service` domain will be allowed to create and manage BPF ring buffers, and load BPF programs
+- Creation of PAM contexts
+    - The `teleport_ssh_service_intermediary` domain will be allowed to `dlopen` the PAM module and call it's exported functions
+    - The location of the PAM module will be configurable
+- TCP port forwarding
+    - The `teleport_ssh_service_networking` domain will be allowed to listen to arbitrary ports and connect to arbitrary network addresses
+- X11 forwarding
+    - The `teleport_ssh_service_networking` domain will be allowed to create and manage X11 sockets
+- SFTP support
+    - The `teleport_ssh_service_intermediary` domain will be allowed to re-execute itself with `sftp` as the first argument
```

Comment:
That's a lot, can we maybe leave less common features support out of initial scope?

Reply:
Updated to only include what some of our clients specifically asked for